RuntimeDetour/Platforms/Runtime/DetourRuntimeNETCore30Platform.cs

```diff
@@ -0,0 +1,446 @@
+                    "Type",
+                    MC.TypeAttributes.Public | MC.TypeAttributes.Abstract
+                ) {
+                    BaseType = module.TypeSystem.Object
+                };
+                module.Types.Add(type);
+
+                MethodDefinition method = new MethodDefinition(
+                    MethodName,
+                    MC.MethodAttributes.Static | MC.MethodAttributes.Public,
+                    module.ImportReference(typeof(Type))
+                ) {
+                    IsInternalCall = true
+                };
+                method.Parameters.Add(new ParameterDefinition(module.ImportReference(typeof(IntPtr))));
+                type.Methods.Add(method);
+
+                assembly = ReflectionHelper.Load(module);
+            }
+
+            MakeAssemblySystemAssembly(assembly);
+
+            return _getTypeFromHandleUnsafeMethod = assembly.GetType("System.Type").GetMethod(MethodName);
+        }
+
+        private static FieldInfo _runtimeAssemblyPtrField = Type.GetType("System.Reflection.RuntimeAssembly").GetField("m_assembly", BindingFlags.Instance | BindingFlags.NonPublic);
+        protected virtual unsafe void MakeAssemblySystemAssembly(Assembly assembly) {
+
+            // RuntimeAssembly.m_assembly is a DomainAssembly*,
+            // which contains an Assembly*,
+            // which contains a PEAssembly*,
+            // which is a subclass of PEFile
+            // which has a `flags` field, with bit 0x01 representing 'system'
+
+            const int PEFILE_SYSTEM = 0x01;
+
+            IntPtr domAssembly = (IntPtr) _runtimeAssemblyPtrField.GetValue(assembly);
+
+            // DomainAssembly in src/coreclr/src/vm/domainfile.h
+            int domOffset =
+                IntPtr.Size + // VTable ptr
+                // DomainFile
+                IntPtr.Size + // PTR_AppDomain               m_pDomain;
+                IntPtr.Size + // PTR_PEFile                  m_pFile;
+                IntPtr.Size + // PTR_PEFile                  m_pOriginalFile;
+                IntPtr.Size + // PTR_Module                  m_pModule;
+                sizeof(int) + // FileLoadLevel               m_level; // FileLoadLevel is an enum with unspecified type; I assume it defaults to 'int' because that's what `enum class` does
+                IntPtr.Size + // LOADERHANDLE                m_hExposedModuleObject;
+                IntPtr.Size + // ExInfo* m_pError;
+                sizeof(int) + // DWORD                    m_notifyflags;
+                sizeof(int) + // BOOL                        m_loading; // no matter the actual size of this BOOL, the next member is a pointer, and we'd always be misaligned
+                IntPtr.Size + // DynamicMethodTable * m_pDynamicMethodTable;
+                IntPtr.Size + // class UMThunkHash *m_pUMThunkHash;
+                sizeof(int) + // BOOL m_bDisableActivationCheck;
+                sizeof(int) + // DWORD m_dwReasonForRejectingNativeImage;
```

Comment:
What happened to the spacing here? 😅

Reply:
Those are the exact lines, complete with spaces, copied directly out of the source. Its really inconsistent in the runtime for some reason.